compiler/rustc_target/src/abi/mod.rs

```diff
@@ -775,10 +775,7 @@ pub enum Abi {
 impl Abi {
     /// Returns `true` if the layout corresponds to an unsized type.
     pub fn is_unsized(&self) -> bool {
-        match *self {
-            Abi::Uninhabited | Abi::Scalar(_) | Abi::ScalarPair(..) | Abi::Vector { .. } => false,
-            Abi::Aggregate { sized } => !sized,
-        }
+        matches!(self, Abi::Aggregate { sized: false })
```

Comment:
I think this was intended to explicitly enumerate all variants, so that we update it when adding new ones?